### ISSUE TITLE
[BUG] Kube-vip image version should be v0.4.2.

### ISSUE THREAD

**Issue by yaocw2020** [harvester/harvester#2033]
**Describe the bug**

Kube-vip image version should be v0.4.2.

**Environment:**
 - Harvester ISO version: v1.0.1 rc1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal with Dell PowerEdge R630

**Additional context**

**Comment by TachunLin** [comment]
Current Kube-vip image version on harvester `master-9285ca26-head` (03/25) is `kube-vip:main` 
https://github.com/kube-vip/kube-vip/pkgs/container/kube-vip

```
harvester-node01-220325:~ # kubectl get pods kube-vip-lgv9q -n harvester-system -o json | grep image
                "image": "ghcr.io/kube-vip/kube-vip:main",
                "imagePullPolicy": "IfNotPresent",
                "image": "ghcr.io/kube-vip/kube-vip:main",
                "imageID": "sha256:1c402444bab2d634b4fb2dccfb8f1652b29e5c8b58daa149e80de82b16c76857",
``` 

According to previous discussion, plan to verify this issue on v1.0.1-rc2 

Test plan would cover:
1.  https://harvester.github.io/tests/manual/network/disable-and-enable-vlan-cluster-network/ 
2.  Install 3 nodes harvester cluster use DHCP VIP
3.  Install 3 nodes harvester cluster use Fixed VIP
4.  Reboot and shutdown cluster nodes scenarios

**Comment by guangbochen** [comment]
We have bumped the `kube-vip` image version to `main` to include a PR https://github.com/kube-vip/kube-vip/pull/383 which will help fix a known issue https://github.com/harvester/harvester/issues/1985. If the VIP team doesn't cut a newer version, then we'll just use the specific master tag `344cb491eb84c0a8dc73e54fa043ebf4ea44bf02` instead.

**Comment by TachunLin** [comment]
Verified fixed on Harvester `v1.0.1-rc2`. Closet this issue.

* Current Kube-vip image version on harvester v1.0.1-rc2 is `kube-vip:main`
```
harvester-node02-101rc2:~ # kubectl get pods kube-vip-8xrvh -n harvester-system -o json | grep image
                "image": "ghcr.io/kube-vip/kube-vip:main",
                "imagePullPolicy": "IfNotPresent",
                "image": "ghcr.io/kube-vip/kube-vip:main",
                "imageID": "sha256:2df9a2468470a6f04920e2c858bafb8bc1d025feeb0d3152504f48365e910fce",
```

### Basic test plan

- [x] **`PASS`** https://harvester.github.io/tests/manual/network/disable-and-enable-vlan-cluster-network/ 
- [x] **`PASS`** Install 3 nodes harvester cluster use DHCP VIP
- [x] **`PASS`** Install 3 nodes harvester cluster use Fixed VIP (use iPXE example)

### Check the DHCP VIP exist under the following scenarios
- [x] **`PASS`** Reboot  all 3 nodes harvester at the same time
- [x] **`PASS`** Reboot only 1 node harvester
- [x] **`PASS`** Shutdown all 3 nodes harvester at the same time and startup after 10 minutes
- [x] **`PASS`** Shutdown only 1 node harvester
  ![image](https://user-images.githubusercontent.com/29251855/161431103-d4802fbb-c328-494e-bc8d-6cf1d7ac112d.png)

### Check the Fixed VIP exist under the following scenarios
- [x] **`PASS`** Reboot  all 3 nodes harvester at the same time
- [x] **`PASS`** Reboot only 1 node harvester
- [x] **`PASS`** Shutdown all 3 nodes harvester at the same time and startup after 10 minutes
- [x] **`PASS`** Shutdown only 1 node harvester
  ![image](https://user-images.githubusercontent.com/29251855/161500540-bb6542cd-f1ba-42ac-846a-9270e3e74347.png)

### 
### Test Information
* Test Environment: 3 nodes harvester on local kvm machines
* Harvester version: v1.0.1-rc2
